state-chain/pallets/cf-funding/src/lib.rs

```diff
@@ -367,7 +343,7 @@ pub mod pallet {
 
 			// Not allowed to redeem if we are an active bidder in the auction phase
 			if T::EpochInfo::is_auction_phase() {
-				ensure!(!ActiveBidder::<T>::get(&account_id), Error::<T>::AuctionPhase);
+				T::BidderProvider::ensure_not_bidding(&account_id)?;
```

Comment:
Instead of linking the BidderProvider I would suggest creating a `trait RedemptionCheck { fn ensure_can_redeem(&account) -> Result<(), DispatchError>; }`. 

Then we can move this whole check into the validator pallet, and we have more flexibility to add other checks in future. And we can also (I think) delete the BidderProvider.

Should also make it easier to mock/test.

Reply:
Yeah the old trait can be removed.